syndesis/src/modules/integrations/pages/IntegrationsPage.tsx

```diff
@@ -120,91 +126,111 @@ export default class IntegrationsPage extends ListViewToolbarAbstractComponent<
-                  }
-                  onUpdateCurrentSortType={this.onUpdateCurrentSortType}
-                >
-                  <WithLoader
-                    error={error}
-                    loading={!hasData}
-                    loaderChildren={
-                      <IntegrationsListSkeleton
-                        width={800}
-                        style={{
-                          backgroundColor: '#FFF',
-                          marginTop: 30,
-                        }}
-                      />
-                    }
-                    errorChildren={<div>TODO</div>}
-                  >
-                    {() => (
-                      <IntegrationsList>
-                        {filteredAndSortedIntegrations.map(
-                          (mi: IntegrationWithMonitoring, index) => (
-                            <IntegrationsListItem
-                              integrationId={mi.integration.id!}
-                              integrationName={mi.integration.name!}
-                              currentState={mi.integration.currentState!}
-                              targetState={mi.integration.targetState!}
-                              isConfigurationRequired={
-                                !!(
-                                  mi.integration!.board!.warnings ||
-                                  mi.integration!.board!.errors ||
-                                  mi.integration!.board!.notices
-                                )
-                              }
-                              monitoringValue={
-                                mi.monitoring &&
-                                mi.monitoring.detailedState.value
-                              }
-                              monitoringCurrentStep={
-                                mi.monitoring &&
-                                mi.monitoring.detailedState.currentStep
-                              }
-                              monitoringTotalSteps={
-                                mi.monitoring &&
-                                mi.monitoring.detailedState.totalSteps
-                              }
-                              key={index}
-                            />
-                          )
-                        )}
-                      </IntegrationsList>
-                    )}
-                  </WithLoader>
-                </IntegrationsListView>
-              );
-            }}
-          </WithConnections>
+        {({ read, data: integrationsData, hasData, error }) => (
+          <ServerEventsContext.Consumer>
+            {({ registerChangeListener, unregisterChangeListener }) => (
+              <WithChangeListener
+                listenerKey="integration-page"
+                read={read}
+                registerChangeListener={registerChangeListener}
+                unregisterChangeListener={unregisterChangeListener}
+                filter={this.changeFilter}
+              >
```

Comment:
Same comment as above, an helper `WithUpdatingMonitoredIntegrations` component would be useful I think.